sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out

```diff
@@ -67,7 +67,7 @@ select interval '2 seconds' / 0
 -- !query schema
 struct<>
 -- !query output
-java.lang.ArithmeticException
+org.apache.spark.SparkArithmeticException
```

Comment:
So we allow changing the exception type. Then can we make `SparkError` a concrete exception class and eventually throw `SparkError` everywhere?

Reply:
My goal here was to create a single trait that exposes the fields we're looking for, such as `sqlState`. By mixing it with base exception types like `java.lang.ArithmeticException` rather than creating a concrete exception class, I'm hoping that we can minimize backwards compatibility issues.